internal/sdkprovider/service/organization/organizational_unit.go

```diff
@@ -158,8 +160,17 @@ func resourceOrganizationalUnitUpdate(ctx context.Context, d *schema.ResourceDat
 }
 
 func resourceOrganizationalUnitDelete(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
-	if err := client.AccountDelete(ctx, d.Id()); common.IsCritical(err) {
-		return err
+	// sometimes the project happens to be deleted with some delay, so we need to retry
+	if err := retry.Do(
+		func() error {
+			return client.AccountDelete(ctx, d.Id())
+		},
+		retry.Attempts(3),
+		retry.Delay(5*time.Second),
+		retry.DelayType(retry.FixedDelay),
+		retry.Context(ctx),
+	); err != nil {
+		return fmt.Errorf("error deleting organizational unit: %w", err)
```

Comment:
Please use `WaitUntilNotFound`. It can be that first request returns something like 5xx, and the second 404. This will keep it retrying.